modules/apps/commerce/commerce-product-definitions-web/src/main/java/com/liferay/commerce/product/definitions/web/internal/frontend/data/set/provider/CommerceProductInstanceOptionsValuesDataProvider.java

```diff
@@ -388,6 +373,39 @@ private JSONArray _addToJSONArray(
 					CP_DEFINITION_OPTION_VALUE_REL_ID_ACCESSOR));
 	}
 
+	private List<CPDefinitionOptionValueRel>
+			_getAllowedCPDefinitionOptionValueRels(
+				CPDefinitionOptionRel cpDefinitionOptionRel,
+				List<CPDefinitionOptionValueRel>
+					selectedCPDefinitionOptionValueRels)
+		throws PortalException {
+
+		List<CPDefinitionOptionValueRel> allowedCPDefinitionOptionValueRels =
+			_filterBySelectedCPDefinitionOptionValueRelIds(
+				cpDefinitionOptionRel, selectedCPDefinitionOptionValueRels);
+
+		if (cpDefinitionOptionRel.isPriceContributor()) {
```

Comment:
I think you can add these two if in the _filterBySelectedCPDefinitionOptionValueRelIds method